react-native-reanimated.d.ts

```diff
@@ -242,7 +242,7 @@ declare module 'react-native-reanimated' {
       getNode(): ReactNativeScrollView;
     }
     export class Code extends Component<CodeProps> {}
-    export function createAnimatedComponent<S extends object, P extends { style?: StyleProp<S>; }>(component: ComponentType<P>): ComponentType<AnimateProps<S, P>>;
+    export function createAnimatedComponent<S extends object, P extends { style?: StyleProp<S>; }>(component: ComponentClass<P>): ComponentClass<AnimateProps<S, P>>;
```

Comment:
```suggestion
    export function createAnimatedComponent<S extends object, P extends object>(component: ComponentClass<P>): ComponentClass<AnimateProps<S, P>>;
```

Reply:
`Animated.createAnimatedComponent(Button);` still does not work. Although I haven't found any use case for this(by that I mean updating animated props of a button on UI) it just indicates that there might be situations like this failing for the valid use cases. This does not work because `ButtonProps` does not contain `style?: StyleProp<object>`. It also won't work for any valid class components which do not contain such prop(`style?: ...`).

Example:

```
interface Props {
  name: string;
  // style?: StyleProp<object>; // will work only with this
}

class MyComponent extends React.Component<Props> {}

Animated.createAnimatedComponent(MyComponent);
```

In order to handle custom components, I'd suggest to change it like I proposed or some other way that would make it work.